codalab/lib/worksheet_util.py

```diff
@@ -80,6 +83,126 @@ def worksheet_line(description, uuid):
     return '[%s]{{%s}}' % (description, uuid)
 
 
+
+############################################################
+
+
+class WorksheetResolver(object):
```

Comment:
We do want to create a new file perhaps called `worksheet_resolver.py`.  We can move `canonicalize.py` into this class and delete `canonicalize.py`.